components/brave_wallet/browser/zcash/rust/BUILD.gn

```diff
@@ -45,6 +45,62 @@ source_set("orchard_impl") {
   ]
 }
 
+rust_static_library("zcash_protocol") {
+  visibility = [ ":zcash_primitives" ]
+  crate_name = "zcash_protocol"
+  crate_root = "librustzcash/components/zcash_protocol/src/lib.rs"
+  allow_unsafe = true
+  sources = [
+    "librustzcash/components/zcash_protocol/src/consensus.rs",
+    "librustzcash/components/zcash_protocol/src/constants.rs",
+    "librustzcash/components/zcash_protocol/src/constants/mainnet.rs",
+    "librustzcash/components/zcash_protocol/src/constants/regtest.rs",
+    "librustzcash/components/zcash_protocol/src/constants/testnet.rs",
+    "librustzcash/components/zcash_protocol/src/lib.rs",
+    "librustzcash/components/zcash_protocol/src/memo.rs",
+    "librustzcash/components/zcash_protocol/src/value.rs",
+  ]
+  deps = [ "//brave/third_party/rust/memuse/v0_2:lib" ]
+}
+
+rust_static_library("zcash_primitives") {
+  visibility = [
+    ":rust_lib",
+    ":rust_lib_cxx_generated",
+  ]
+  crate_name = "zcash_primitives"
+  crate_root = "librustzcash/zcash_primitives/src/lib.rs"
+  allow_unsafe = true
+  sources = [
+    "librustzcash/zcash_primitives/src/lib.rs",
+    "librustzcash/zcash_primitives/src/merkle_tree.rs",
+    "librustzcash/zcash_primitives/src/transaction/components.rs",
+    "librustzcash/zcash_primitives/src/transaction/components/orchard.rs",
+    "librustzcash/zcash_primitives/src/transaction/mod.rs",
+  ]
+  deps = [
+    ":zcash_encoding",
+    ":zcash_protocol",
+    "//brave/third_party/rust/byteorder/v1:lib",
+    "//brave/third_party/rust/incrementalmerkletree/v0_5:lib",
+    "//brave/third_party/rust/memuse/v0_2:lib",
+    "//brave/third_party/rust/nonempty/v0_7:lib",
+    "//brave/third_party/rust/orchard/v0_7:lib",
+  ]
+}
+
+rust_static_library("zcash_encoding") {
+  visibility = [ ":zcash_primitives" ]
+  crate_name = "zcash_encoding"
+  crate_root = "librustzcash/components/zcash_encoding/src/lib.rs"
+  allow_unsafe = true
```

Comment:
same as above, this should normally only be needed when using cxx

Reply:
zcash_encoding fails without this :
```
error[E0453]: deny(unsafe_code) incompatible with previous forbid
 --> ../../brave/components/brave_wallet/browser/zcash/rust/librustzcash/src/components/zcash_encoding/src/lib.rs:9:9
  |
9 | #![deny(unsafe_code)]
  |         ^^^^^^^^^^^ overruled by previous forbid
```